app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -193,9 +196,12 @@ class ComponentsValidator(
   def validF(implicit ec: ExecutionContext) =
     ().valid.pure[Future]
 
-  private def validateDate(fieldValue: FormComponent, date: Date): ValidatedType[Unit] = {
+  private def validateDate(
+    fieldValue: FormComponent,
+    date: Date,
+    otherFieldValue: Option[FormComponent]): ValidatedType[Unit] = {
     val reqFieldValidResult = validateDateRequiredField(fieldValue)
-    val otherRulesValidResult = validateDateImpl(fieldValue, date)(data)
+    val otherRulesValidResult = validateDateImpl(fieldValue, date, otherFieldValue)(data)
     Monoid[ValidatedType[Unit]].combineAll(List(reqFieldValidResult, otherRulesValidResult))
```

Comment:
Maybe we could inline the two variables passed as argument as they are only used one, so you could remove brackets too saving four lines of code ;-)

Reply:
changes were resolved